pipeline/pipelines/core_generation_pipeline.py

```diff
@@ -55,13 +55,10 @@ class _GoCoreTaskFactory(code_gen.TaskFactoryBase):
     """
 
     def get_tasks(self, **kwargs):
-        return [
-            protoc_tasks.ProtoCodeGenTask('CoreProtoGen',
-                                          inject=kwargs),
-            protoc_tasks.GoExtractImportBaseTask('ExtractGoPackageName',
-                                                 inject=kwargs),
-            protoc_tasks.GoLangUpdateImportsTask('UpdateImports',
-                                                 inject=kwargs)]
+        tasks = [protoc_tasks.ProtoCodeGenTask,
+                 protoc_tasks.GoExtractImportBaseTask,
+                 protoc_tasks.GoLangUpdateImportsTask]
+        return task_utils.instantiate_tasks(tasks, kwargs)
```

Comment:
That is what I had originally. However, there is a problem when implementing this in the batch pipeline, because the get_tasks function for the BatchTaskFactory doesn't return only tasks - it returns flows which contains tasks, so these must necessarily be instantiated. While it would be possible to change BatchTaskFactory.get_tasks to return a long list of all the tasks, this would lose the structure of api x language tasks, and would prevent changing from a linear flow to an unordered flow in the future.

One thing that could be done in core_generation_pipeline is to add a base class that does the instantiation, like gapic_generation_pipeline and grpc_generation_pipeline